workers/unity/Packages/com.improbable.gdk.core/Dynamic/Dynamic.cs

```diff
@@ -0,0 +1,77 @@
+using System;
+using System.Collections.Generic;
+using Improbable.Worker.Core;
+using Unity.Entities;
+
+namespace Improbable.Gdk.Core
+{
+    public delegate T ComponentDeserializer<out T>(ComponentData data, World world) where T : ISpatialComponentData;
+
+    public delegate T UpdateDeserializer<out T>(ComponentUpdate update, World world) where T : ISpatialComponentUpdate;
+
+    public static class Dynamic
+    {
+        public interface IHandler
+        {
+            void Accept<TData, TUpdate>(uint componentId,
+                ComponentDeserializer<TData> deserializeComponentData,
+                UpdateDeserializer<TUpdate> deserializeComponentUpdate)
+                where TData : ISpatialComponentData
+                where TUpdate : ISpatialComponentUpdate;
+        }
+
+        private static readonly Dictionary<uint, IDynamicInvokable> idsToComponents =
+            new Dictionary<uint, IDynamicInvokable>();
+
+        private static readonly Dictionary<Type, uint> componentsToIds = new Dictionary<Type, uint>();
+
+        static Dynamic()
+        {
+            foreach (var assembly in AppDomain.CurrentDomain.GetAssemblies())
```

Comment:
Drop some `foreach` nesting and tests: `AppDomain.CurrentDomain.GetAssemblies().SelectMany(assembly => assembly.GetTypes()).Where(type => !type.IsAbstract)`